oak_client/src/lib.rs

```diff
@@ -22,34 +22,92 @@ pub mod proto {
 pub mod transport;
 pub mod verifier;
 
-use anyhow::Context;
+use crate::transport::AsyncEvidenceProvider;
+use anyhow::{anyhow, Context};
+use oak_crypto::{
+    schema::{AeadEncryptedMessage, HpkeRequest, HpkeResponse},
+    SenderCryptoProvider,
+};
+use prost::Message;
 use std::vec::Vec;
 
+const EMPTY_ASSOCIATED_DATA: &[u8] = b"";
+
 /// Client for connecting to Oak.
 /// Represents a Relying Party from the RATS Architecture:
 /// <https://www.rfc-editor.org/rfc/rfc9334.html#name-relying-party>
 pub struct OakClient<T>
 where
-    T: micro_rpc::AsyncTransport<Error = anyhow::Error>,
+    T: micro_rpc::AsyncTransport<Error = anyhow::Error> + AsyncEvidenceProvider,
```

Comment:
Why is this needed here? AFAICT it's only needed to create the client, and never after that, so it can just be an additional bound for the create method?